runtime/src/consensus/transaction.rs

```diff
@@ -0,0 +1,37 @@
+use crate::common::{crypto::signature::Signed, quantity::Quantity};
+
+pub const SIGNATURE_CONTEXT: &[u8] = b"oasis-core/consensus: tx for chain ";
+
+/// Transaction is an unsigned consensus transaction.
+#[derive(Debug, Default, cbor::Encode, cbor::Decode)]
+pub struct Transaction {
+    // Nonce is a nonce to prevent replay.
+    pub nonce: u64,
+    // Fee is an optional fee that the sender commits to pay to execute this
+    // transaction.
+    pub fee: Fee,
+
+    // Method is the method that should be called.
+    pub method: MethodName,
+    // Body is the method call body.
+    pub body: Vec<u8>,
```

Comment:
The (more generally) correct type is `cbor::Value` not `Vec<u8>`. The `Vec<u8>` only works for this specific `registry.ProveFreshness` transaction as it takes a binary blob as body.